pcdsdevices/gasdet.py

```diff
@@ -0,0 +1,345 @@
+    )
+    rate = FCpt(
+        EpicsSignalRO,
+        '{self.prefix}:{self.module}:MEventRate',
+        kind='config'
+    )
+    num_events = FCpt(
+        EpicsSignalRO,
+        '{self.prefix}:{self.module}:MEventCount',
+        kind='config'
+    )
+    num_trig_timeouts = FCpt(
+        EpicsSignalRO,
+        '{self.prefix}:{self.module}:MTriggerTimeouts',
+        kind='config'
+    )
+    num_truncated = FCpt(
+        EpicsSignalRO,
+        '{self.prefix}:{self.module}:MTruncated',
+        kind='config'
+    )
+    acq_type = FCpt(
+        EpicsSignal,
+        '{self.prefix}:{self.module}:MType',
+        kind='config',
+        string=True
+    )
+    acq_mode = FCpt(
+        EpicsSignal,
+        '{self.prefix}:{self.module}:MMode',
+        kind='config',
+        string=True
+    )
+    acq_mode_flags = FCpt(
+        EpicsSignal,
+        '{self.prefix}:{self.module}:MModeFlags',
+        kind='config',
+        string=True
+    )
+    clock_type = FCpt(
+        EpicsSignalRO,
+        '{self.prefix}:{self.module}:MClockType',
+        kind='config',
+        string=True
+    )
+    crate_number = FCpt(
+        EpicsSignalRO,
+        '{self.prefix}:{self.module}:MCrateNb',
+        kind='config'
+    )
+    nADC_bits = FCpt(
+        EpicsSignalRO,
+        '{self.prefix}:{self.module}:MNbrADCBits',
+        kind='config'
+    )
+    crate_slot = FCpt(
+        EpicsSignalRO,
+        '{self.prefix}:{self.module}:MPosInCrate',
+        kind='config'
+    )
+    temp = FCpt(
+        EpicsSignalRO,
+        '{self.prefix}:{self.module}:MTemperature_m',
+        kind='config'
+    )
+    input_freq = FCpt(
+        EpicsSignalRO,
+        '{self.prefix}:{self.module}:MInputFrequency',
+        kind='config'
+    )
+
+    def __init__(self,
+                 prefix,
+                 module,
+                 platform=None,
+                 name='acqiris_module',
+                 **kwargs):
+        self.module = module
+        self.prefix = prefix
+        self.platform = platform
+        if len(self.module) != 3:
+            raise ValueError("acqiris 'module'"
+                             + " expects a 3-digit module identifier,"
+                             + " e.g. '240'")
+        self._mod_pref = self.module[:2]
+        super().__init__(prefix, name=name, **kwargs)
+
+    def start(self):
+        """
+        Start the acqiris DAQ
+        """
+        self.run_state.put(0)
+
+    def stop(self):
+        """
+        Stop the acqiris DAQ
+        """
+        self.run_state.put(1)
+
+
+class gasdet(Device):
+    """
+    Class to read out pulse energy from a PMT/diode as digitized
+    by an acqiris in the GasDetDAQ EPICS IOC.
+
+    Parameters
+    ----------
+    prefix : ``str``
+        The PV base of the digitizer.
+    platform : ``str``
+        This a legacy PV segment found in the GasDetDAQ IOC.
+        For example, '202' applies to
+        the HXR gas detector system... for some reason.
```

Comment:
Isn't this platform just the old FEE location PV segment? E.g. a device with 202 would be about 20.2% down the length of the FEE. Sort of like a street address.

Reply:
Like I think 202 is part of the prefix in reality, and then the channels all have different prefixes that differ by one number

Reply:
202 is the old fee numeric address system. 2** was the upstream GEM area code. It's being dropped, and 202 should go away.